test/integration/scheduler/workload_controller_test.go

```diff
@@ -69,6 +72,138 @@ var _ = ginkgo.Describe("Workload controller with scheduler", func() {
 		updatedCQ = kueue.ClusterQueue{}
 	})
 
+	ginkgo.When("the queue has admission check strategies", func() {
+		var (
+			flavor1           *kueue.ResourceFlavor
+			flavor2           *kueue.ResourceFlavor
+			check1            *kueue.AdmissionCheck
+			check2            *kueue.AdmissionCheck
+			check3            *kueue.AdmissionCheck
+			reservationFlavor string = "reservation"
+			updatedWl         kueue.Workload
+			flavorOnDemand    string              = "on-demand"
+			resourceGPU       corev1.ResourceName = "example.com/gpu"
+		)
+
+		ginkgo.BeforeEach(func() {
+			flavor1 = testing.MakeResourceFlavor(flavorOnDemand).Obj()
+			gomega.Expect(k8sClient.Create(ctx, flavor1)).Should(gomega.Succeed())
+
+			flavor2 = testing.MakeResourceFlavor(reservationFlavor).Obj()
+			gomega.Expect(k8sClient.Create(ctx, flavor2)).Should(gomega.Succeed())
+
+			check1 = testing.MakeAdmissionCheck("check1").ControllerName("ctrl1").Obj()
+			gomega.Expect(k8sClient.Create(ctx, check1)).Should(gomega.Succeed())
+			util.SetAdmissionCheckActive(ctx, k8sClient, check1, metav1.ConditionTrue)
+
+			check2 = testing.MakeAdmissionCheck("check2").ControllerName("ctrl2").Obj()
+			gomega.Expect(k8sClient.Create(ctx, check2)).Should(gomega.Succeed())
+			util.SetAdmissionCheckActive(ctx, k8sClient, check2, metav1.ConditionTrue)
+
+			check3 = testing.MakeAdmissionCheck("check3").ControllerName("ctrl3").Obj()
+			gomega.Expect(k8sClient.Create(ctx, check3)).Should(gomega.Succeed())
+			util.SetAdmissionCheckActive(ctx, k8sClient, check3, metav1.ConditionTrue)
+
+			clusterQueue = testing.MakeClusterQueue("cluster-queue").
+				AdmissionCheckStrategy(
+					*testing.MakeAdmissionCheckStrategy("check1", flavorOnDemand).Obj(),
+					*testing.MakeAdmissionCheckStrategy("check2").Obj(),
+					*testing.MakeAdmissionCheckStrategy("check3", reservationFlavor).Obj()).
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(reservationFlavor).Resource(resourceGPU, "1", "1").Obj(),
+					*testing.MakeFlavorQuotas(flavorOnDemand).Resource(resourceGPU, "5", "5").Obj()).
+				Cohort("cohort").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+
+			localQueue = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check3, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check2, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check1, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, flavor1, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, flavor2, true)
+		})
+
+		ginkgo.FIt("the workload should have appropriate AdditionalChecks added", func() {
```

Comment:
```suggestion
		ginkgo.It("the workload should have appropriate AdditionalChecks added", func() {
```
Isn't this for the debug?

Reply:
Fixed